### ISSUE TITLE
Tiled isn't showing the the tiles despite the JSON file loading correctly in-game

### ISSUE THREAD

**Issue by Archeia** [bjorn/tiled#1871]
My team and I are collaborating on maps and we are not using .tmx format. However, when I open the JSON file we have in tiled, it doesn't display the map at all. Here's an example:
![image](https://user-images.githubusercontent.com/6222499/35670529-cfa5ce70-0773-11e8-9467-83382be845cf.png)

As you can see it's only showing gray things despite the tilesets loaded on the sidebar. Is there anyway I can send this to a private email or something?

**Comment by bjorn** [comment]
> As you can see it's only showing gray things despite the tilesets loaded on the sidebar. Is there anyway I can send this to a private email or something?

Yes, you can e-mail the files causing this issue to bjorn@lindeijer.nl.



> EDIT:
> For some reason, on specific maps, Tiled doesn't like it if the source is displayed like this:
> "source":"../../../Archeia/OMORI/maps/collision.json"
> 
> But if I change it to this format, it suddenly works ok.
> "source":"collision.json"

Do you have any idea how the source reference ended up that way? Normally such references should work fine, provided the `collision.json` file is available at that path relative to the map file.

Besides, when Tiled does have trouble finding such a file, it should result in a box showing right below the file tab listing the files that could not be found. Did such a thing pop up when opening the map? I think I should put some kind of status icon in the status bar in this case, to keep it available even when you close it.

**Comment by Archeia** [comment]
Hi @bjorn 

All of the json map files and the tileset json files are in the same folder. We share our map files through dropbox. I'm on Windows PC and one is on Mac. So when the "tilesets are missing" in the Mac computer, the user has to redirect them again to the dropbox folder. Even when editing the map outside a dropbox folder.

**Comment by bjorn** [comment]
@Archeia Hmm, this is a tricky situation. So you share your maps and tilesets in Dropbox, but you mention that you're "editing the map outside a Dropbox folder".

When you save a map outside of the Dropbox folder that is using a tileset that is inside the Dropbox folder, then Tiled stores a relative path in the map file that directs it to the tileset in the Dropbox folder. But when you then copy that map file into the Dropbox folder, this relative path will no longer work. To avoid issues you should either:

* Have all files outside of Dropbox while working with Tiled. Only use Dropbox to share the directory containing the maps and tilesets with others, but don't open maps or tilesets directly from the Dropbox folder in Tiled. This way, the relative paths will stay valid.

* Don't have any files outside of Dropbox. Just work with the files directly in Dropbox. To avoid issues with saving files directly into Dropbox, you should disable the "Use safe writing of files" option in the Preferences.

If you're going to have files both inside and outside of Dropbox, you're always at risk of saving a map to Dropbox that refers to tilesets outside of Dropbox, or saving a map outside of Dropbox that refers to a tileset inside Dropbox.

**Comment by bjorn** [comment]
Though, that doesn't explain why Tiled doesn't show the "Some files count not be found" message. You are using the latest version?

**Comment by Archeia** [comment]
@bjorn sorry I think I said this wrong. What we do is edit our project in dropbox and this error still happens. So we even attempted to move it outside of dropbox to see if it will fix the issues. 

> Don't have any files outside of Dropbox. Just work with the files directly in Dropbox. To avoid issues with saving files directly into Dropbox, you should disable the "Use safe writing of files" option in the Preferences.

This is exactly what we're doing. And yes, I'm using the latest itch.io version. The problem is that the person in MacOSX has a different directory for dropbox than someone who is in Windows. I'm guessing that's part of the problem?
